### ISSUE TITLE
Compile error when switching from 3.5.1-next.1 to 3.5.1-next.2 or 3.5.1

### ISSUE THREAD

**Issue by mrt181** [riok/mapperly#1264]
**Describe the bug**
When building with `3.5.1` or `3.5.1-next.2` I get
* CS8081
* CS8604

Building with `3.5.1-next.1` works.

```bash
> dotnet-outdated -inc riok
» stow.Infrastructure
  [net8.0]
  Riok.Mapperly  3.5.1-next.1 -> 3.5.1

Version color legend:
<red>   : Major version update or pre-release version. Possible breaking changes.
<yellow>: Minor version update. Backwards-compatible features added.
<green> : Patch version update. Backwards-compatible bug fixes.
                                                                                                                                                                                                                                                                                                                        You can upgrade packages to the latest version by passing the -u or -u:prompt option.
Elapsed: 00:00:02.5563750

> dotnet build src/stow.Infrastructure/stow.Infrastructure.csproj
MSBuild version 17.9.8+b34f75857 for .NET
  Determining projects to restore...
  Restored /home/martin/repos/stow/src/stow.UseCases/stow.UseCases.csproj (in 353 ms).
  Restored /home/martin/repos/stow/src/stow.Core/stow.Core.csproj (in 353 ms).
  Restored /home/martin/repos/stow/src/stow.Infrastructure/stow.Infrastructure.csproj (in 353 ms).
  stow.Core -> /home/martin/repos/stow/src/stow.Core/bin/Debug/net8.0/stow.Core.dll
  stow.UseCases -> /home/martin/repos/stow/src/stow.UseCases/bin/Debug/net8.0/stow.UseCases.dll
  stow.Infrastructure -> /home/martin/repos/stow/src/stow.Infrastructure/bin/Debug/net8.0/stow.Infrastructure.dll

Build succeeded.
    0 Warning(s)
    0 Error(s)
                                                                                                                                                                                                                                                                                                                        Time Elapsed 00:00:03.31

> dotnet-outdated -inc riok -u
» stow.Infrastructure
  [net8.0]
  Riok.Mapperly  3.5.1-next.1 -> 3.5.1

Version color legend:
<red>   : Major version update or pre-release version. Possible breaking changes.
<yellow>: Minor version update. Backwards-compatible features added.
<green> : Patch version update. Backwards-compatible bug fixes.

Upgrading package Riok.Mapperly...
Project stow.Infrastructure [net8.0] upgraded successfully

Elapsed: 00:00:03.5719592

> dotnet build src/stow.Infrastructure/stow.Infrastructure.csproj
MSBuild version 17.9.8+b34f75857 for .NET
  Determining projects to restore...
  All projects are up-to-date for restore.
  stow.Core -> /home/martin/repos/stow/src/stow.Core/bin/Debug/net8.0/stow.Core.dll
  stow.UseCases -> /home/martin/repos/stow/src/stow.UseCases/bin/Debug/net8.0/stow.UseCases.dll
/home/martin/repos/stow/src/stow.Infrastructure/Riok.Mapperly/Riok.Mapperly.MapperGenerator/SwingMapper.g.cs(19,154): error CS8081: Expression does not have a name. [/home/martin/repos/stow/src/stow.Infrastructure/stow.Infrastructure.csproj]
/home/martin/repos/stow/src/stow.Infrastructure/Riok.Mapperly/Riok.Mapperly.MapperGenerator/SwingMapper.g.cs(20,108): error CS8081: Expression does not have a name. [/home/martin/repos/stow/src/stow.Infrastructure/stow.Infrastructure.csproj]
/home/martin/repos/stow/src/stow.Infrastructure/Riok.Mapperly/Riok.Mapperly.MapperGenerator/SwingMapper.g.cs(21,108): error CS8081: Expression does not have a name. [/home/martin/repos/stow/src/stow.Infrastructure/stow.Infrastructure.csproj]
/home/martin/repos/stow/src/stow.Infrastructure/Riok.Mapperly/Riok.Mapperly.MapperGenerator/SwingMapper.g.cs(23,168): error CS8081: Expression does not have a name. [/home/martin/repos/stow/src/stow.Infrastructure/stow.Infrastructure.csproj]
/home/martin/repos/stow/src/stow.Infrastructure/Riok.Mapperly/Riok.Mapperly.MapperGenerator/SwingMapper.g.cs(19,58): error CS8604: Possible null reference argument for parameter 'nominationDayAheadDeadline' in 'TimeOnly SwingMapper.MapTimeOnly(string nominationDayAheadDeadline)'. [/home/martin/repos/stow/src/stow.Infrastructure/stow.Infrastructure.csproj]
/home/martin/repos/stow/src/stow.Infrastructure/Riok.Mapperly/Riok.Mapperly.MapperGenerator/SwingMapper.g.cs(20,29): error CS8601: Possible null reference assignment. [/home/martin/repos/stow/src/stow.Infrastructure/stow.Infrastructure.csproj]
/home/martin/repos/stow/src/stow.Infrastructure/Riok.Mapperly/Riok.Mapperly.MapperGenerator/SwingMapper.g.cs(21,29): error CS8601: Possible null reference assignment. [/home/martin/repos/stow/src/stow.Infrastructure/stow.Infrastructure.csproj]
/home/martin/repos/stow/src/stow.Infrastructure/Riok.Mapperly/Riok.Mapperly.MapperGenerator/SwingMapper.g.cs(23,78): error CS8604: Possible null reference argument for parameter 'renominationLeadTime' in 'TimeSpan SwingMapper.MapTimeSpan(string renominationLeadTime)'. [/home/martin/repos/stow/src/stow.Infrastructure/stow.Infrastructure.csproj]

Build FAILED.

/home/martin/repos/stow/src/stow.Infrastructure/Riok.Mapperly/Riok.Mapperly.MapperGenerator/SwingMapper.g.cs(19,154): error CS8081: Expression does not have a name. [/home/martin/repos/stow/src/stow.Infrastructure/stow.Infrastructure.csproj]
/home/martin/repos/stow/src/stow.Infrastructure/Riok.Mapperly/Riok.Mapperly.MapperGenerator/SwingMapper.g.cs(20,108): error CS8081: Expression does not have a name. [/home/martin/repos/stow/src/stow.Infrastructure/stow.Infrastructure.csproj]
/home/martin/repos/stow/src/stow.Infrastructure/Riok.Mapperly/Riok.Mapperly.MapperGenerator/SwingMapper.g.cs(21,108): error CS8081: Expression does not have a name. [/home/martin/repos/stow/src/stow.Infrastructure/stow.Infrastructure.csproj]
/home/martin/repos/stow/src/stow.Infrastructure/Riok.Mapperly/Riok.Mapperly.MapperGenerator/SwingMapper.g.cs(23,168): error CS8081: Expression does not have a name. [/home/martin/repos/stow/src/stow.Infrastructure/stow.Infrastructure.csproj]
/home/martin/repos/stow/src/stow.Infrastructure/Riok.Mapperly/Riok.Mapperly.MapperGenerator/SwingMapper.g.cs(19,58): error CS8604: Possible null reference argument for parameter 'nominationDayAheadDeadline' in 'TimeOnly SwingMapper.MapTimeOnly(string nominationDayAheadDeadline)'. [/home/martin/repos/stow/src/stow.Infrastructure/stow.Infrastructure.csproj]
/home/martin/repos/stow/src/stow.Infrastructure/Riok.Mapperly/Riok.Mapperly.MapperGenerator/SwingMapper.g.cs(20,29): error CS8601: Possible null reference assignment. [/home/martin/repos/stow/src/stow.Infrastructure/stow.Infrastructure.csproj]
/home/martin/repos/stow/src/stow.Infrastructure/Riok.Mapperly/Riok.Mapperly.MapperGenerator/SwingMapper.g.cs(21,29): error CS8601: Possible null reference assignment. [/home/martin/repos/stow/src/stow.Infrastructure/stow.Infrastructure.csproj]
/home/martin/repos/stow/src/stow.Infrastructure/Riok.Mapperly/Riok.Mapperly.MapperGenerator/SwingMapper.g.cs(23,78): error CS8604: Possible null reference argument for parameter 'renominationLeadTime' in 'TimeSpan SwingMapper.MapTimeSpan(string renominationLeadTime)'. [/home/martin/repos/stow/src/stow.Infrastructure/stow.Infrastructure.csproj]
    0 Warning(s)
    8 Error(s)

Time Elapsed 00:00:03.12
```

**Declaration code**
```csharp
[Mapper]
public partial class SwingMapper : IMapper<Swing, Core.Data.Swing>
{
  [UseMapper]
  private readonly LegMapper _legMapper = new();

  [MapperIgnoreSource(nameof(Swing.Headers))]
  [MapperIgnoreSource(nameof(Swing.RowOffset))]
  [MapperIgnoreSource(nameof(Swing.RowPartition))]
  [MapProperty(nameof(Swing.RowTime), nameof(Core.Data.Swing.Timestamp))]
  [MapProperty($"{nameof(Swing.Asset)}.{nameof(Swing.Asset.EndurId)}", nameof(Core.Data.Swing.EndurId))]
  [MapProperty($"{nameof(Swing.Asset)}.{nameof(Swing.Asset.ContractualLimits)}", nameof(Core.Data.Swing.ContractualLimits))]
  [MapProperty(
    $"{nameof(Swing.Asset)}.{nameof(Swing.Asset.ExternalContractName)}",
    nameof(Core.Data.Swing.ExternalContractName)
  )]
  [MapProperty(
    $"{nameof(Swing.Asset)}.{nameof(Swing.Asset.HolidayCalendarType)}",
    nameof(Core.Data.Swing.HolidayCalendarType)
  )]
  [MapProperty($"{nameof(Swing.Asset)}.{nameof(Swing.Asset.Legs)}", nameof(Core.Data.Swing.Legs))]
  [MapProperty($"{nameof(Swing.Asset)}.{nameof(Swing.Asset.Name)}", nameof(Core.Data.Swing.Name))]
  [MapProperty(
    $"{nameof(Swing.Asset)}.{nameof(Swing.Asset.NominationCalendarType)}",
    nameof(Core.Data.Swing.NominationCalendarType)
  )]
  [MapProperty(
    $"{nameof(Swing.Asset)}.{nameof(Swing.Asset.NominationDayAheadDeadline)}",
    nameof(Core.Data.Swing.NominationDayAheadDeadline),
    Use = nameof(MapTimeOnly)
  )]
  [MapProperty($"{nameof(Swing.Asset)}.{nameof(Swing.Asset.Portfolio)}", nameof(Core.Data.Swing.Portfolio))]
  [MapProperty($"{nameof(Swing.Asset)}.{nameof(Swing.Asset.Reference)}", nameof(Core.Data.Swing.Reference))]
  [MapProperty(
    $"{nameof(Swing.Asset)}.{nameof(Swing.Asset.RenominationAllowed)}",
    nameof(Core.Data.Swing.RenominationAllowed)
  )]
  [MapProperty(
    $"{nameof(Swing.Asset)}.{nameof(Swing.Asset.RenominationLeadTime)}",
    nameof(Core.Data.Swing.RenominationLeadTime),
    Use = nameof(MapTimeSpan)
  )]
  [MapProperty(
    $"{nameof(Swing.Asset)}.{nameof(Swing.Asset.RenominationMaxCount)}",
    nameof(Core.Data.Swing.RenominationMaxCount)
  )]
  [MapProperty($"{nameof(Swing.Asset)}.{nameof(Swing.Asset.AssetType)}", nameof(Core.Data.Swing.AssetType))]
  public partial Core.Data.Swing Map(Swing swing);
  }
}
```

**Actual relevant generated code**
With version `5.3.1-next.1`|
```csharp
// <auto-generated />
#nullable enable
namespace stow.Infrastructure.Mapper
{
    public partial class SwingMapper
    {
        [global::System.CodeDom.Compiler.GeneratedCode("Riok.Mapperly", "3.5.1.0")]
        public partial global::stow.Core.Data.Swing Map(global::stow.Infrastructure.Data.Swing swing)
        {
            var target = new global::stow.Core.Data.Swing(swing.Id ?? throw new System.ArgumentNullException(nameof(swing.Id)), swing.DealConclusion != null ? global::System.DateTimeOffset.Parse(swing.DealConclusion) : throw new System.ArgumentNullException(nameof(swing.DealConclusion)), swing.Counterparty ?? throw new System.ArgumentNullException(nameof(swing.Counterparty)), swing.DealOwner ?? throw new System.ArgumentNullException(nameof(swing.DealOwner)), (global::stow.Core.Data.Direction)swing.Direction, swing.End != null ? global::System.DateTimeOffset.Parse(swing.End) : throw new System.ArgumentNullException(nameof(swing.End)), swing.Start != null ? global::System.DateTimeOffset.Parse(swing.Start) : throw new System.ArgumentNullException(nameof(swing.Start)), swing.TraderId ?? throw new System.ArgumentNullException(nameof(swing.TraderId)), swing.RowTime, swing.Asset != null ? MapToAsset(swing.Asset) : throw new System.ArgumentNullException(nameof(swing.Asset)))
            {
                EndurId = swing.Asset?.EndurId ?? throw new System.ArgumentNullException(nameof(swing.Asset.EndurId)),
                ContractualLimits = swing.Asset != null ? MapToContractualLimitArray(swing.Asset.ContractualLimits) : throw new System.ArgumentNullException(nameof(swing.Asset.ContractualLimits)),
                ExternalContractName = swing.Asset?.ExternalContractName ?? throw new System.ArgumentNullException(nameof(swing.Asset.ExternalContractName)),
                HolidayCalendarType = swing.Asset != null ? (global::stow.Core.Data.HolidayCalendarType)swing.Asset.HolidayCalendarType : throw new System.ArgumentNullException(nameof(swing.Asset.HolidayCalendarType)),
                Legs = swing.Asset != null ? MapToLegArray(swing.Asset.Legs) : throw new System.ArgumentNullException(nameof(swing.Asset.Legs)),
                Name = swing.Asset?.Name ?? throw new System.ArgumentNullException(nameof(swing.Asset.Name)),
                NominationCalendarType = swing.Asset != null ? (global::stow.Core.Data.NominationCalendarType)swing.Asset.NominationCalendarType : throw new System.ArgumentNullException(nameof(swing.Asset.NominationCalendarType)),
                NominationDayAheadDeadline = swing.Asset != null ? MapTimeOnly(swing.Asset.NominationDayAheadDeadline) : throw new System.ArgumentNullException(nameof(swing.Asset.NominationDayAheadDeadline)),
                Portfolio = swing.Asset?.Portfolio ?? throw new System.ArgumentNullException(nameof(swing.Asset.Portfolio)),
                Reference = swing.Asset?.Reference ?? throw new System.ArgumentNullException(nameof(swing.Asset.Reference)),
                RenominationAllowed = swing.Asset?.RenominationAllowed ?? throw new System.ArgumentNullException(nameof(swing.Asset.RenominationAllowed)),
                RenominationLeadTime = swing.Asset != null ? MapTimeSpan(swing.Asset.RenominationLeadTime) : default(global::System.TimeSpan?),
                RenominationMaxCount = swing.Asset?.RenominationMaxCount,
                AssetType = swing.Asset != null ? (global::stow.Core.Data.AssetType)swing.Asset.AssetType : throw new System.ArgumentNullException(nameof(swing.Asset.AssetType)),
            };
            return target;
        }

        [global::System.CodeDom.Compiler.GeneratedCode("Riok.Mapperly", "3.5.1.0")]
        private global::stow.Core.Data.Asset MapToAsset(global::stow.Infrastructure.Data.Asset source)
        {
            var target = new global::stow.Core.Data.Asset(source.EndurId, MapToContractualLimitArray(source.ContractualLimits), source.ExternalContractName, (global::stow.Core.Data.HolidayCalendarType)source.HolidayCalendarType, MapToLegArray(source.Legs), source.Name, (global::stow.Core.Data.NominationCalendarType)source.NominationCalendarType, MapTimeOnly(source.NominationDayAheadDeadline), source.Portfolio, source.Reference, source.RenominationAllowed, MapTimeSpan(source.RenominationLeadTime), source.RenominationMaxCount, (global::stow.Core.Data.AssetType)source.AssetType);
            return target;
        }

        [global::System.CodeDom.Compiler.GeneratedCode("Riok.Mapperly", "3.5.1.0")]
        private global::stow.Core.Data.ContractualLimit MapToContractualLimit(global::stow.Infrastructure.Data.ContractualLimit source)
        {
            var target = new global::stow.Core.Data.ContractualLimit((global::stow.Core.Data.ContractualLimitType)source.Type, source.Id, source.Location, global::System.DateTimeOffset.Parse(source.Start), global::System.DateTimeOffset.Parse(source.End), (global::stow.Core.Data.LimitGranularity)source.Granularity, source.Limit, (global::stow.Core.Data.MinMax)source.MinMax, (global::stow.Core.Data.FirmInterruptible)source.FirmInterruptible, (global::stow.Core.Data.Unit)source.Unit);
            return target;
        }

        [global::System.CodeDom.Compiler.GeneratedCode("Riok.Mapperly", "3.5.1.0")]
        private global::stow.Core.Data.ContractualLimit[] MapToContractualLimitArray(global::stow.Infrastructure.Data.ContractualLimit[] source)
        {
            var target = new global::stow.Core.Data.ContractualLimit[source.Length];
            for (var i = 0; i < source.Length; i++)
            {
                target[i] = MapToContractualLimit(source[i]);
            }
            return target;
        }

        [global::System.CodeDom.Compiler.GeneratedCode("Riok.Mapperly", "3.5.1.0")]
        private global::stow.Core.Data.Leg[] MapToLegArray(global::stow.Infrastructure.Data.Leg[] source)
        {
            var target = new global::stow.Core.Data.Leg[source.Length];
            for (var i = 0; i < source.Length; i++)
            {
                target[i] = _legMapper.Map(source[i]);
            }
            return target;
        }
    }
}
```
With version `5.3.1`|
```csharp
// <auto-generated />
#nullable enable
namespace stow.Infrastructure.Mapper
{
    public partial class SwingMapper
    {
        [global::System.CodeDom.Compiler.GeneratedCode("Riok.Mapperly", "3.5.1.0")]
        public partial global::stow.Core.Data.Swing Map(global::stow.Infrastructure.Data.Swing swing)
        {
            var target = new global::stow.Core.Data.Swing(swing.Id ?? throw new System.ArgumentNullException(nameof(swing.Id)), swing.DealConclusion != null ? global::System.DateTimeOffset.Parse(swing.DealConclusion) : throw new System.ArgumentNullException(nameof(swing.DealConclusion)), swing.Counterparty ?? throw new System.ArgumentNullException(nameof(swing.Counterparty)), swing.DealOwner ?? throw new System.ArgumentNullException(nameof(swing.DealOwner)), (global::stow.Core.Data.Direction)swing.Direction, swing.End != null ? global::System.DateTimeOffset.Parse(swing.End) : throw new System.ArgumentNullException(nameof(swing.End)), swing.Start != null ? global::System.DateTimeOffset.Parse(swing.Start) : throw new System.ArgumentNullException(nameof(swing.Start)), swing.TraderId ?? throw new System.ArgumentNullException(nameof(swing.TraderId)), swing.RowTime, swing.Asset != null ? MapToAsset(swing.Asset) : throw new System.ArgumentNullException(nameof(swing.Asset)))
            {
                EndurId = swing.Asset?.EndurId ?? throw new System.ArgumentNullException(nameof(swing.Asset.EndurId)),
                ContractualLimits = swing.Asset != null ? MapToContractualLimitArray(swing.Asset.ContractualLimits) : throw new System.ArgumentNullException(nameof(swing.Asset.ContractualLimits)),
                ExternalContractName = swing.Asset?.ExternalContractName ?? throw new System.ArgumentNullException(nameof(swing.Asset.ExternalContractName)),
                HolidayCalendarType = swing.Asset != null ? (global::stow.Core.Data.HolidayCalendarType)swing.Asset.HolidayCalendarType : throw new System.ArgumentNullException(nameof(swing.Asset.HolidayCalendarType)),
                Legs = swing.Asset != null ? MapToLegArray(swing.Asset.Legs) : throw new System.ArgumentNullException(nameof(swing.Asset.Legs)),
                Name = swing.Asset?.Name ?? throw new System.ArgumentNullException(nameof(swing.Asset.Name)),
                NominationCalendarType = swing.Asset != null ? (global::stow.Core.Data.NominationCalendarType)swing.Asset.NominationCalendarType : throw new System.ArgumentNullException(nameof(swing.Asset.NominationCalendarType)),
                NominationDayAheadDeadline = MapTimeOnly(swing.Asset?.NominationDayAheadDeadline == null ? throw new System.ArgumentNullException(nameof(swing.Asset?.NominationDayAheadDeadline)) : swing.Asset?.NominationDayAheadDeadline),
                Portfolio = swing.Asset?.Portfolio == null ? throw new System.ArgumentNullException(nameof(swing.Asset?.Portfolio)) : swing.Asset?.Portfolio,
                Reference = swing.Asset?.Reference == null ? throw new System.ArgumentNullException(nameof(swing.Asset?.Reference)) : swing.Asset?.Reference,
                RenominationAllowed = swing.Asset?.RenominationAllowed ?? throw new System.ArgumentNullException(nameof(swing.Asset.RenominationAllowed)),
                RenominationLeadTime = (global::System.TimeSpan?)MapTimeSpan(swing.Asset?.RenominationLeadTime == null ? throw new System.ArgumentNullException(nameof(swing.Asset?.RenominationLeadTime)) : swing.Asset?.RenominationLeadTime),
                RenominationMaxCount = swing.Asset?.RenominationMaxCount,
                AssetType = swing.Asset != null ? (global::stow.Core.Data.AssetType)swing.Asset.AssetType : throw new System.ArgumentNullException(nameof(swing.Asset.AssetType)),
            };
            return target;
        }

        [global::System.CodeDom.Compiler.GeneratedCode("Riok.Mapperly", "3.5.1.0")]
        private global::stow.Core.Data.Asset MapToAsset(global::stow.Infrastructure.Data.Asset source)
        {
            var target = new global::stow.Core.Data.Asset(source.EndurId, MapToContractualLimitArray(source.ContractualLimits), source.ExternalContractName, (global::stow.Core.Data.HolidayCalendarType)source.HolidayCalendarType, MapToLegArray(source.Legs), source.Name, (global::stow.Core.Data.NominationCalendarType)source.NominationCalendarType, MapTimeOnly(source.NominationDayAheadDeadline), source.Portfolio, source.Reference, source.RenominationAllowed, MapTimeSpan(source.RenominationLeadTime), source.RenominationMaxCount, (global::stow.Core.Data.AssetType)source.AssetType);
            return target;
        }

        [global::System.CodeDom.Compiler.GeneratedCode("Riok.Mapperly", "3.5.1.0")]
        private global::stow.Core.Data.ContractualLimit MapToContractualLimit(global::stow.Infrastructure.Data.ContractualLimit source)
        {
            var target = new global::stow.Core.Data.ContractualLimit((global::stow.Core.Data.ContractualLimitType)source.Type, source.Id, source.Location, global::System.DateTimeOffset.Parse(source.Start), global::System.DateTimeOffset.Parse(source.End), (global::stow.Core.Data.LimitGranularity)source.Granularity, source.Limit, (global::stow.Core.Data.MinMax)source.MinMax, (global::stow.Core.Data.FirmInterruptible)source.FirmInterruptible, (global::stow.Core.Data.Unit)source.Unit);
            return target;
        }

        [global::System.CodeDom.Compiler.GeneratedCode("Riok.Mapperly", "3.5.1.0")]
        private global::stow.Core.Data.ContractualLimit[] MapToContractualLimitArray(global::stow.Infrastructure.Data.ContractualLimit[] source)
        {
            var target = new global::stow.Core.Data.ContractualLimit[source.Length];
            for (var i = 0; i < source.Length; i++)
            {
                target[i] = MapToContractualLimit(source[i]);
            }
            return target;
        }

        [global::System.CodeDom.Compiler.GeneratedCode("Riok.Mapperly", "3.5.1.0")]
        private global::stow.Core.Data.Leg[] MapToLegArray(global::stow.Infrastructure.Data.Leg[] source)
        {
            var target = new global::stow.Core.Data.Leg[source.Length];
            for (var i = 0; i < source.Length; i++)
            {
                target[i] = _legMapper.Map(source[i]);
            }
            return target;
        }
    }
}
```

This is the diff between the generated code
```diff
--- 3.5.1-next.1.cs	2024-04-30 14:19:35.633922538 +0200
+++ 3.5.1.cs	2024-04-30 14:21:07.433921954 +0200
@@ -16,11 +16,11 @@
                 Legs = swing.Asset != null ? MapToLegArray(swing.Asset.Legs) : throw new System.ArgumentNullException(nameof(swing.Asset.Legs)),
                 Name = swing.Asset?.Name ?? throw new System.ArgumentNullException(nameof(swing.Asset.Name)),
                 NominationCalendarType = swing.Asset != null ? (global::stow.Core.Data.NominationCalendarType)swing.Asset.NominationCalendarType : throw new System.ArgumentNullException(nameof(swing.Asset.NominationCalendarType)),
-                NominationDayAheadDeadline = swing.Asset != null ? MapTimeOnly(swing.Asset.NominationDayAheadDeadline) : throw new System.ArgumentNullException(nameof(swing.Asset.NominationDayAheadDeadline)),
-                Portfolio = swing.Asset?.Portfolio ?? throw new System.ArgumentNullException(nameof(swing.Asset.Portfolio)),
-                Reference = swing.Asset?.Reference ?? throw new System.ArgumentNullException(nameof(swing.Asset.Reference)),
+                NominationDayAheadDeadline = MapTimeOnly(swing.Asset?.NominationDayAheadDeadline == null ? throw new System.ArgumentNullException(nameof(swing.Asset?.NominationDayAheadDeadline)) : swing.Asset?.NominationDayAheadDeadline),
+                Portfolio = swing.Asset?.Portfolio == null ? throw new System.ArgumentNullException(nameof(swing.Asset?.Portfolio)) : swing.Asset?.Portfolio,
+                Reference = swing.Asset?.Reference == null ? throw new System.ArgumentNullException(nameof(swing.Asset?.Reference)) : swing.Asset?.Reference,
                 RenominationAllowed = swing.Asset?.RenominationAllowed ?? throw new System.ArgumentNullException(nameof(swing.Asset.RenominationAllowed)),
-                RenominationLeadTime = swing.Asset != null ? MapTimeSpan(swing.Asset.RenominationLeadTime) : default(global::System.TimeSpan?),
+                RenominationLeadTime = (global::System.TimeSpan?)MapTimeSpan(swing.Asset?.RenominationLeadTime == null ? throw new System.ArgumentNullException(nameof(swing.Asset?.RenominationLeadTime)) : swing.Asset?.RenominationLeadTime),
                 RenominationMaxCount = swing.Asset?.RenominationMaxCount,
                 AssetType = swing.Asset != null ? (global::stow.Core.Data.AssetType)swing.Asset.AssetType : throw new System.ArgumentNullException(nameof(swing.Asset.AssetType)),
             };
@@ -64,3 +64,4 @@
         }
     }
 }
+

```
**Expected relevant generated code**
```csharp
// The generated code how you expect it to look like
```

**Environment (please complete the following information):**
- Mapperly Version: [3.5.1-next.1, 3.5.1-next.2, 3.5.1]
- Nullable reference types: [enabled]
- .NET Version: [.NET 8.0.204]
- Target Framework: [.net8.0]
- Compiler Version: [4.9.2-3.24129.6 (9934fb9e)]
- C# Language Version: [12.0]
- IDE: [Rider 2024.1 and nvim 0.9.4 + omnisharp 1.39.11]
- OS: [WSL 2.1.5.0, Ubuntu 22.04.4 LTS]

**Additional context**

**Comment by latonz** [comment]
You are referring to the switch from 3.5.1-next.1 to 3.5.1 correct? I think there is a typo in the version number in the title and in the issue body.

**Comment by mrt181** [comment]
Yes, of course. Updated the version numbers

**Comment by latonz** [comment]
As far as I can tell from the error message this should be fixed with `3.6.0-next.1`, could you try this version?

**Comment by mrt181** [comment]
Works with `3.6.0-next.1`
